### PR TITLE
Rework build pathname functions

### PR DESCRIPTION
**Pull request by Ni3znajomy**
This PR makes ```build_pathname*``` functions use AMTL's ```ke::path::Format``` and ```ke::path::FormatVa```. Code is shorter and cleaner inside those functions now.

Also editor removed white spaces in edited files, hope you don't mind that 😇, but if you do, i'll create commit without those unrelated changes.

### CODE REVIEW

File: amxmodx/amxmodx.h [hunk 1]
```diff
@@ -105,10 +105,8 @@ extern AMX_NATIVE_INFO g_GameConfigNatives[];
 
 #if defined(_WIN32)
```
Comment (Arkshine) [inline amxmodx/amxmodx.h]: You can probably remove those defines now, we have a `PLATFORM_` version.

**Comment by Ni3znajomy** [comment]
Commit divided into smaller ones. Now it should be easier to review.

File: amxmodx/amxmodx.h [hunk 2]
```diff
@@ -96,19 +96,17 @@ extern AMX_NATIVE_INFO g_GameConfigNatives[];
...
-#if defined(_WIN32)
+#if defined PLATFORM_WINDOWS
 	typedef HINSTANCE DLHANDLE;
 #else
 	typedef void* DLHANDLE;
 	#define INFINITE 0xFFFFFFFF
 #endif
 
-#if defined(_WIN32)
+#if defined PLATFORM_WINDOWS
```
Comment (Arkshine) [inline amxmodx/amxmodx.h]: As said in the previous review, you can remove those constants, they are redundant with `PLATFORM_` version

Comment (Ni3znajomy) [inline amxmodx/amxmodx.h]: They're gone now

**Comment by Arkshine** [comment]
Alright, it should be okay now, thanks!

**Pull request merged**